### ISSUE TITLE
misleading error message

### ISSUE THREAD

**Issue by antiphasis** [benhoyt/scandir#54]
Setup to reproduce: using windows, python27 and the current version of scandir.  
Create a few directories, one with german umlauts, e.g Aufträge.

```shell
mkdir c:\devel\playground\auftraege
mkdir c:\devel\playground\aufträge
```

Now let scandir work with the directories

```shell
C:\>python -c "import scandir;list(scandir.walk('c:/devel/playground'))"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python27\lib\site-packages\scandir.py", line 654, in walk
    for entry in walk(new_path, topdown, onerror, followlinks):
  File "C:\Python27\lib\site-packages\scandir.py", line 594, in walk
    scandir_it = scandir(top)
TypeError: os.scandir() doesn't support bytes path on Windows, use Unicode instead
```

When searching scandir.py for the error message: It's thrown at 
- line 147, which belongs to checking for PY3 on Win32. 
```python
if IS_PY3 and sys.platform == 'win32':
    def scandir_generic(path=u'.'):
        if isinstance(path, bytes):
            raise TypeError("os.scandir() doesn't support bytes path on Windows, use Unicode instead")
        return _scandir_generic(path)
    scandir_generic.__doc__ = _scandir_generic.__doc__
else:
    scandir_generic = _scandir_generic
```
- line 381, which belongs to checking if PY3 inside the whole if WIN32 
```python
        if IS_PY3:
            def scandir_python(path=u'.'):
                if isinstance(path, bytes):
                    raise TypeError("os.scandir() doesn't support bytes path on Windows, use Unicode instead")
                return _scandir_python(path)
            scandir_python.__doc__ = _scandir_python.__doc__
        else:
            scandir_python = _scandir_python
```

**Comment by benhoyt** [comment]
Good find. I believe this is being raised by `_scandir.c`, however, [this code](https://github.com/benhoyt/scandir/blob/3bb47fb2404dfe920dc63fa46767c07c7baf3c93/_scandir.c#L1718). I *think* that on Python 2.x on Windows this should be just silently failing and convert the special characters to ? characters. This is kind of broken behaviour, but it's what os.listdir() does. I'll look into this next time I work on scandir. What you want to do is pass in a unicode string for the scandir path argument.

**Comment by antiphasis** [comment]
Yepp, when passing the path as unicode it's working fine.

**Comment by antiphasis** [comment]
Almost forgotten: Haven't had these issues with the older versions (approx. a year ago).

**Comment by benhoyt** [comment]
I finally fixed this in https://github.com/benhoyt/scandir/commit/489cfa77f83dc9f3f5535d6b0f696ea1bf7e9b8c. From the commit message:

> I could have fixed this to exactly mimic os.walk()'s behaviour on Windows Python 2.x, but that's really broken: ASCII directories get treated as directories, some unicode directories go in the "dirs" list, some unicode directory names go into the "files" list and appear with non-ASCII chars replace with ASCII '?'. So I decided to fix it to what you want 99% of the time: if you pass a bytes path into walk() on Windows Python 2.x, it'll convert it to unicode so the walk does what you expect. This does mean that the names returned are all unicode strings, but that's almost always fine and what you want too.

**Comment by benhoyt** [comment]
FYI, I'll release a new version on PyPI shortly.

**Comment by benhoyt** [comment]
This is fixed now in [version 1.2](https://github.com/benhoyt/scandir/releases/tag/v1.2).
